specification/data-semantic-conventions.md

```diff
@@ -14,49 +14,105 @@ This way, the operator will not need to learn specifics of a language and
 telemetry collected from multi-language micro-service can still be easily
 correlated and cross-analyzed.
 
-## HTTP client
+## HTTP Spans
 
-This span type represents an outbound HTTP request.
-
-For a HTTP client span, `SpanKind` MUST be `Client`.
+This span types represents HTTP requests. They can be used for http and https
+schemes and various HTTP versions like 1.1, 2 and SPDY.
 
-Given an [RFC 3986](https://www.ietf.org/rfc/rfc3986.txt) compliant URI of the form
+Given an [RFC 3986](https://tools.ietf.org/html/rfc3986) compliant URI of the form
 `scheme:[//authority]path[?query][#fragment]`, the span name of the span SHOULD
 be set to to the URI path value.
 
-If a framework can identify a value that represents the identity of the request
-and has a lower cardinality than the URI path, this value MUST be used for the span name instead.
+If a value that represents the identity of the request
+and has a lower cardinality than the URI path can be identified, this value MUST
+be used for the span name instead.
 
 | Attribute name | Notes and examples                                           | Required? |
 | :------------- | :----------------------------------------------------------- | --------- |
 | `component`    | Denotes the type of the span and needs to be `"http"`. | Yes |
 | `http.method` | HTTP request method. E.g. `"GET"`. | Yes |
-| `http.url` | HTTP URL of this request, represented as `scheme://host:port/path?query#fragment` E.g. `"https://example.com:779/path/12314/?q=ddds#123"`. | Yes |
-| `http.status_code` | [HTTP response status code](https://tools.ietf.org/html/rfc7231). E.g. `200` (integer) | No |
-| `http.status_text` | [HTTP reason phrase](https://www.ietf.org/rfc/rfc2616.txt). E.g. `"OK"` | No |
+| `http.url` | Full HTTP request URL in the form `scheme://host:port/path?query#fragment`. Usually the fragment is not transmitted over HTTP, but if it is known, it should be included nevertheless. | Defined later. |
+| `http.status_code` | [HTTP response status code][]. E.g. `200` (integer) | No |
+| `http.status_text` | [HTTP reason phrase][]. E.g. `"OK"` | No |
+| `http.flavor` | Kind of HTTP protocol used: `"1.0"`, `"1.1"`, `"2"`, `"SPDY"` or `"QUIC"`. |  If not TCP-based (`QUIC`). |
```

Comment:
Do we also want an `http.user_agent` field like in OpenCensus (https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md#attributes)?

Or other "UEM" related attributes like referrer URL, a visitor ID or session ID? And would they make sense on the client side too (common attribute) or only on the server side?

Reply:
Existing OTel implementations (e..g for ASP.NET) provide `http.user_agent`.

Reply:
well user-agent is just a header. I would go for a more generic approach of request/response header capturing.